### PR TITLE
feat: add ability to lint/check of links in documentation locally

### PR DESCRIPTION
**Pull request by mavogel**
- implements #40 
- adds linters for the docs and also formatting for the consisting `hcl` code
- adds local make target to have a better/faster workflow, and also the ability for auto-fixing
- also added an additional/enhanced website linter for spellcheck and formatting (so we have 2 atm)

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -7,6 +7,15 @@ default: build
 build: fmtcheck
 	go install
 
+setup:
+	go mod download
+	cd tools && GO111MODULE=on go install github.com/client9/misspell/cmd/misspell
+	cd tools && GO111MODULE=on go install github.com/katbyte/terrafmt
+	cd tools && GO111MODULE=on go install github.com/golangci/golangci-lint/cmd/golangci-lint
```
Comment (suzuki-shunsuke) [inline GNUmakefile:14]: https://golangci-lint.run/usage/install/#install-from-source

> Note: such go get installation aren't guaranteed to work. We recommend using binary installation.

Comment (suzuki-shunsuke) [reply GNUmakefile:14]: There are two options.

1. Install golangci-lint with the official script
1. Leave the installation to developers

## Install golangci-lint with the official script

https://golangci-lint.run/usage/install/#other-ci

```
$ curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.33.0
```

In that case, `curl` is required. And we should update the version of golangci-lint with [Renovate](https://docs.renovatebot.com/) or something.

Comment (suzuki-shunsuke) [reply GNUmakefile:14]: Oh, sorry.
I understood. https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

File: scripts/markdown-link-check.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+if [ -z "$(docker image list -q markdown-link-check)" ]; then
+  echo "This script requires the markdown-link-check Docker image."
+  echo ""
+  echo "More information: https://github.com/tcort/markdown-link-check"
+  echo ""
+  echo "Steps: "
+  echo "git clone git@github.com:tcort/markdown-link-check.git"
+  echo "cd markdown-link-check"
+  echo "docker build -t markdown-link-check ."
```
Comment (suzuki-shunsuke) [inline scripts/markdown-link-check.sh:14]: https://github.com/tcort/markdown-link-check#run-using-docker

How about using `ghcr.io/tcort/markdown-link-check:stable`?

Comment (mavogel) [reply scripts/markdown-link-check.sh]: Nice :) Last time I failed to pull this image 🤔 Anyway, it works now. Fixed 803f268

**Review by suzuki-shunsuke — APPROVED** [review]

**Pull request merged**